common/domain/handler.go

```diff
@@ -669,6 +658,111 @@ func (d *handlerImpl) DeprecateDomain(
 	return nil
 }
 
+// UpdateIsolationGroups is used for draining and undraining of isolation-groups for a domain.
+// Like the isolation-group API, this controller expects Upsert semantics for
+// isolation-groups and does not modify any other domain information.
+//
+// Isolation-groups are regional in their configuration scope, so it's expected that this upsert
+// includes configuration for both clusters every time.
+//
+// The update is handled like other domain updates in that they expected to be replicated. So
+// unlike the global isolation-group API it shouldn't be necessary to call
+func (d *handlerImpl) UpdateIsolationGroups(
```

Comment:
The refactoring is not a good idea. It introduces duplicate logic and the most important thing is that replication is using `UpdateDomain`